### PR TITLE
Removed IMouseDevice from service locator

### PR DESCRIPTION
**Pull request by kekekeks**
Part of #1013

### CODE REVIEW

File: src/Avalonia.Input/IInputDevice.cs [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+
 namespace Avalonia.Input
 {
     public interface IInputDevice
     {
+        /// <summary>
+        /// Processes raw event. Is called after preprocessing by InputManager
+        /// </summary>
+        /// <param name="ev"></param>
+        void ProcessOwnRawEvent(RawInputEventArgs ev);
```
Comment (grokys) [inline src/Avalonia.Input/IInputDevice.cs]: I think this should probably just be called `ProcessRawEvent`.

File: src/Gtk/Avalonia.Gtk/TopLevelImpl.cs [hunk 2]
```diff
@@ -114,6 +114,8 @@ public Avalonia.Controls.WindowState WindowState
 
         public Action Closed { get; set; }
 
+        public IMouseDevice MouseDevice => GtkMouseDevice.Instance;
```
Comment (grokys) [inline src/Gtk/Avalonia.Gtk/TopLevelImpl.cs]: Could we move this below the `Action` properties to keep things tidy?

File: src/Gtk/Avalonia.Gtk3/WindowBaseImpl.cs [hunk 3]
```diff
@@ -242,6 +242,9 @@ public Size MaxClientSize
 
         public Action Activated { get; set; }
         public Action Closed { get; set; }
+
+        public IMouseDevice MouseDevice => Gtk3Platform.Mouse;
```
Comment (grokys) [inline src/Gtk/Avalonia.Gtk3/WindowBaseImpl.cs]: Could we move this below the `Action` properties to keep things tidy?

File: src/Linux/Avalonia.LinuxFramebuffer/FramebufferToplevelImpl.cs [hunk 4]
```diff
@@ -63,6 +63,9 @@ public void SetCursor(IPlatformHandle cursor)
         public Action<Size> Resized { get; set; }
         public Action<double> ScalingChanged { get; set; }
         public Action Closed { get; set; }
+
+        public IMouseDevice MouseDevice => LinuxFramebufferPlatform.MouseDevice;
```
Comment (grokys) [inline src/Linux/Avalonia.LinuxFramebuffer/FramebufferToplevelImpl.cs]: Could we move this below the `Action` properties to keep things tidy?

File: src/Windows/Avalonia.Win32/WindowImpl.cs [hunk 5]
```diff
@@ -49,6 +49,8 @@ public WindowImpl()
 
         public Action Closed { get; set; }
 
+        public IMouseDevice MouseDevice => WindowsMouseDevice.Instance;
```
Comment (grokys) [inline src/Windows/Avalonia.Win32/WindowImpl.cs]: Could we move this below the Action properties to keep things tidy?

**Review by grokys — CHANGES_REQUESTED** [review]
Looks good to me in general - just a few naming/style nitpicks. Also I'm wondering if we should do the same with `KeyboardDevice` here - it'd be strange to have one but not the other done like this.

**Comment by grokys** [comment]
Also could you add a description of why this needs doing to the PR so that people looking back in 5 years time can see the reasons things ended up how they did?

**Comment by kekekeks** [comment]
I haven't touched keyboard device yet, since it will require changes at least in focus management. This PR is already huge, so it's better to do in a different one

**Pull request merged**